bin/cron/update_pledge_map

```diff
@@ -26,34 +26,60 @@ CENSUS_FORM_KINDS = "'Census2017','HocCensus2017'".freeze
 begin
   CSV.open(file.path, 'w') do |csv|
+    DASHBOARD_DB[:census_submissions].
+      select_append(:census_submissions__id___census_submissions_id).
+      join(:census_submissions_school_infos, census_submission_id: :id).
+      join(:school_infos, id: :school_info_id).
+      join(:schools, id: :school_id).
+      paged_each(rows_per_fetch: 10_000) do |submission|
+      pledged = submission[:pledged]
+      teaches_yes = ["SOME", "ALL"]
+      teaches_cs = teaches_yes.include?(submission[:how_many_10_hours]) || teaches_yes.include?(submission[:how_many_20_hours])
+
+      # Only add to the map if pledge checkbox is checked or the school teaches some form of CS.
+      next unless pledged || teaches_cs
+
+      type = pledged ? "pledged" : "teaches_cs"
+
+      next unless submission[:school_id]
+      row_id = "CENSUS_SUBMISSIONS-#{submission[:census_submissions_id]}"
+      row = [row_id, submission[:name], submission[:zip], submission[:city], type]
+      csv << row
+    end
+
     DB_READONLY[:forms].where("kind in (#{CENSUS_FORM_KINDS})").paged_each(rows_per_fetch: 10_000) do |form|
-      data = JSON.parse(form[:data])
+      # Skip any rows that were migrated to CENSUS_SUBMISSIONS since those were already processed above.
+      form_map = DASHBOARD_DB[:census_submission_form_maps].where(form_id: form[:id])
+      if form_map.empty?
+        data = JSON.parse(form[:data])
 
-      # Was a pledge made?
-      pledged = ["on", "1"].any? {|s| s.casecmp(data["pledge_b"].to_s).zero?}
+        # Was a pledge made?
+        pledged = ["on", "1"].any? {|s| s.casecmp(data["pledge_b"].to_s).zero?}
 
-      # Does the school teach some kind of CS?
-      teaches_cs = ["some", "all"].any? {|s| s.casecmp(data["ten_hours_s"].to_s).zero?} ||
-          ["some", "all"].any? {|s| s.casecmp(data["twenty_hours_s"].to_s).zero?}
+        # Does the school teach some kind of CS?
+        teaches_cs = ["some", "all"].any? {|s| s.casecmp(data["ten_hours_s"].to_s).zero?} ||
+                     ["some", "all"].any? {|s| s.casecmp(data["twenty_hours_s"].to_s).zero?}
 
-      # Only add to the map if pledge checkbox is checked or the school teaches some form of CS.
-      next unless pledged || teaches_cs
+        # Only add to the map if pledge checkbox is checked or the school teaches some form of CS.
+        next unless pledged || teaches_cs
+
+        # Use the school ID in the form.
+        school_id = data["school_id_s"] || data["nces_school_s"]
 
-      # Use the school ID in the form.
-      school_id = data["school_id_s"] || data["nces_school_s"]
+        # Retrieve the ZIP and city for this school district from the dashboard's database.
+        school = DASHBOARD_DB[:schools].where(id: school_id).first
+        next unless school
 
-      # Retrieve the ZIP and city for this school district from the dashboard's database.
-      school = DASHBOARD_DB[:schools].where(id: school_id).first
-      next unless school
+        if pledged
```

Comment:
worth reducing to a one-line ternary?

Reply:
I didn't modify any of the code for handling FORMS rows other than to skip ones that had been moved to census_submissions. That had the side effect of indenting everything, making a nasty-looking diff. 

That said, I did use the one-line ternary in the census_submissions processing (line 42)